presto-main/src/main/java/com/facebook/presto/sql/gen/JoinFilterFunctionCompiler.java

```diff
@@ -112,7 +114,14 @@ public JoinFilterFunctionFactory compileJoinFilterFunction(RowExpression filter,
     private JoinFilterFunctionFactory internalCompileFilterFunctionFactory(RowExpression filterExpression, int leftBlocksSize, Optional<SortExpression> sortChannel)
     {
         Class<? extends InternalJoinFilterFunction> internalJoinFilterFunction = compileInternalJoinFilterFunction(filterExpression, leftBlocksSize);
-        return new IsolatedJoinFilterFunctionFactory(internalJoinFilterFunction, sortChannel);
+
+        List<Class<? extends InternalJoinFilterFunction>> internalInequalityJoinFilterConjuncts = new ArrayList<>();
```

Comment:
What about using `Optional.map()`

```java
List<Class<? extends InternalJoinFilterFunction>> internalInequalityJoinFilterConjuncts = 
   sortChannel.map(channel -> channel.getInequalityJoinFilterConjuncts().stream()
                                  .map(rowExpression -> compileInternalJoinFilterFunction(rowExpression, leftBlocksSize))
                                  .collect(toImmutableList()))
              .orElse(ImmutableMap.of());
``` 

If you do not like this please at least use `ImmutableMap.of()` instead `new ArrayList()`. We use `ImmutableMap.of()` as empty list by convention.